fakestorage/server.go

```diff
@@ -211,9 +212,22 @@ func newServer(options Options) (*Server, error) {
 	return &s, nil
 }
 
+func unescapeMuxVars(vars map[string]string) map[string]string {
```

Comment:
In some places we use `r.URL.Query()`, which provide already decoded values. But `mux.Vars(r)` provide values "as is" (without decoding). That is why we need decode `objectName` and `bucketName` ourself. This is important, because before accessing to storage we encode values and this lead to double encoding.